File: ui-testing/pages/checkout_step_one_page.py
from pages.base_page import BasePage
from selenium.webdriver.common.by import By


class CheckoutStepOne(BasePage):
    CHECKOUT_TITLE = (By.CLASS_NAME, "title")
    FIRST_NAME = (By.ID, 'first-name')
    LAST_NAME = (By.ID, 'last-name')
    ZIP_CODE = (By.ID, 'postal-code')
    CONTINUE_BTN = (By.ID, 'continue')

    def fill_checkout_info(self, first_name, last_name, zip_code):
        self.fill(self.FIRST_NAME, first_name)
        self.fill(self.LAST_NAME, last_name)
        self.fill(self.ZIP_CODE, zip_code)

    def click_continue(self):
        self.click(self.CONTINUE_BTN)

File: playwright-testing/pages/checkout_step_one_page.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './base.page';


export class CheckoutStepOne extends BasePage{ 
    readonly checkoutTitle : Locator;
    readonly firstName : Locator;
    readonly lastName : Locator;
    readonly zipCode : Locator;
    readonly continueBtn : Locator;



    constructor(page : Page){
        super(page);
        this.checkoutTitle = page.locator('.title');
        this.firstName = page.locator('#first-name');
        this.lastName = page.locator('#last-name');
        this.zipCode = page.locator('#postal-code');
        this.continueBtn = page.locator('#continue');
    }


    async fillCheckoutInformation(firstName: string, lastName: string, zipCode: string):Promise<void>{
        await this.firstName.fill(firstName);
        await this.lastName.fill(lastName);
        await this.zipCode.fill(zipCode);
    }   
    async clickContinue(): Promise<void>{
        await this.continueBtn.click();
    }
}